pkg/controller/node/cidr_allocator.go

```diff
@@ -122,7 +129,24 @@ func NewCIDRRangeAllocator(client clientset.Interface, clusterCIDR *net.IPNet, s
 	return ra, nil
 }
 
+func (r *rangeAllocator) leaseNode(nodeName string) bool {
```

Comment:
nit: I understand your motivation of calling it "lease", but to be honest, it would be much simpler to follow if you call those methods:
"insertNodeInProcessing"
"deleteNodeInProcessing"

WDYT?